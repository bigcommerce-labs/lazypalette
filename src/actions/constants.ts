export const enum ConfigUpdateAction {
    PREVIEW = 'PREVIEW',
    SAVE = 'SAVE',
    PUBLISH =  'PUBLISH',
    UPDATE = 'UPDATE',
}

export const enum ToastMessages {
    ErrorSwitchToThemeEditor = 'There was an error switching to the old Theme Editor.',
    ErrorHistory = 'There was an error switching to the selected Theme History entry.',
    ErrorPreview = 'There was an error previewing your changes.',
    ErrorPublish = 'There was an error publishing your changes.',
    ErrorReset = 'There was an error restoring the original theme styles.',
    ErrorSave = 'There was an error saving your changes.',
    ErrorUpdate = 'There was an error applying your update.',
    ErrorVariation = 'There was an error switching styles.',
    Publish = 'Changes have been published.',
    Reset = 'Original theme styles restored.',
    Save = 'Changes have been saved.',
    Undo = 'Changes have been undone.',
    Update = 'Theme has been updated.',
}

export const enum ToastType {
    Success = 'Success',
    Error = 'Error',
}

export const enum ToastTimeout {
    Duration = 6000,
}
