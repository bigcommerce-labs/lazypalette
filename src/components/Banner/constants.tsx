import React from 'react';

export const CopyMessage = 'Link copied to clipboard.';

export const TooltipMessage =
    <>
        <p>You must launch your store from the Getting Started page in order to make it public.</p>
        <p><a href="/manage/getting-started">Go to Getting Started</a></p>
        <p>You can still let other people see your store before it is launched by sharing the private store link.</p>
    </>;

export const MaintenanceMessage = 'Your store is down for maintenance.';
export const PrelaunchMessage = 'Your store has not launched and is not publicly visible.';
