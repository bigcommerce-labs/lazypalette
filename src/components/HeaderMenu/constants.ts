export const enum HeaderMenuLinks {
    CONTROL_PANEL = 'manage',
}

export const enum HeaderMenuMessages {
    GO_BACK = 'Go back to Control Panel',
}
