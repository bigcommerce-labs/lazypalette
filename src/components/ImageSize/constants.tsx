export const enum Messages {
    Dimension = 'Specify a maximum image size for desktop display. We\'ll automatically ' +
        'scale down the image for smaller devices.',
    Lock = 'Constrain Proportions',
}

export const ImageDimensionPattern = (/^[\d]+x[\d]+$/);
