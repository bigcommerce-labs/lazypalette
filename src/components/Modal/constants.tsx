export const enum Messages {
    Reset = 'This theme has unpublished changes. Do you want to proceed?',
    Variation = 'This variation has unpublished changes. Do you want to proceed?',
}
