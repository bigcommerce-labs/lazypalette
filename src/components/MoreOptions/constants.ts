export const RestoreOriginalText = {
    action: 'Restore',
    body: 'All changes to this theme will be restored back to its default settings. ' +
    'This will not publish changes to your live store.',
    title: 'Restore original theme style',
};
