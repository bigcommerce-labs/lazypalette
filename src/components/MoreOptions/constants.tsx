import React from 'react';

import { Link, ModalParagraph } from './styles';

export const CopyThemeText = {
    action: 'Go to My Themes',
    body: `You must make a copy of your theme to edit the theme files.`,
    title: 'Edit Theme Files',
};

const designPolicyUrl = 'https://support.bigcommerce.com/articles/Public/BigCommerce-Design-Policy';

export const EditThemeFilesText = {
    action: 'Edit Theme Files',
    bodyActive:
        <>
            <ModalParagraph>
                It looks like you're about to modify your active theme by editing its theme files. Any changes you apply
                to your active theme will be made to your live store.
            </ModalParagraph>
            <ModalParagraph>
                Please note that BigCommerce is unable to offer support for any theme that has been modified with Edit
                Theme Files.&nbsp;
                <Link href={designPolicyUrl} target="_blank">
                    Learn more
                </Link>
            </ModalParagraph>
        </>,
    bodyInactive:
        <ModalParagraph>
            It looks like you're about to modify your theme by editing its theme files. Please note that BigCommerce
            is unable to offer support for any theme that has been modified with Edit Theme Files.&nbsp;
            <Link href={designPolicyUrl} target="_blank">
                Learn more
            </Link>
        </ModalParagraph>,
    title: 'Edit Theme Files',
};

export const RestoreOriginalText = {
    action: 'Restore',
    body: `All changes to this theme will be restored back to its default settings. This will not publish changes to
        your live store.`,
    title: 'Restore original theme style',
};

export enum CurrentModal {
    COPY_THEME = 'COPY_THEME',
    EDIT_THEME_FILES = 'EDIT_THEME_FILES',
    NONE = 'NONE',
    RESET = 'RESET',
}

export const FileEditorPath = (versionId: string, variationId: string, configurationId: string) =>
    `/manage/file-editor/${versionId}/${variationId}/${configurationId}`;

export const MyThemesPath = '/manage/storefront-manager/my-themes';
