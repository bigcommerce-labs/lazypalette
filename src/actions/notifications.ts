import { Dispatch } from 'redux';

import { State } from '../reducers/reducers';

import { Action } from './action';
import { ToastTimeout } from './constants';

export enum NotificationActionTypes {
    OPEN_NOTIFICATION = 'OPEN_NOTIFICATION',
    CLOSE_NOTIFICATION = 'CLOSE_NOTIFICATION',
}

export interface OpenNotificationAction extends Action  {
    payload: NotificationsProps;
    type: NotificationActionTypes.OPEN_NOTIFICATION;
}

export interface CloseNotificationAction extends Action  {
    type: NotificationActionTypes.CLOSE_NOTIFICATION;
}

export interface NotificationsProps {
    autoDismiss: boolean;
    message: string;
    type: string;
}

export function openNotification(content: NotificationsProps): OpenNotificationAction {
    return {
        payload: {
            autoDismiss: content.autoDismiss,
            message: content.message,
            type: content.type,
        },
        type: NotificationActionTypes.OPEN_NOTIFICATION,
    };
}

export function closeNotification(): CloseNotificationAction {
    return {
        type: NotificationActionTypes.CLOSE_NOTIFICATION,
    };
}

export function createNotification(autoDismiss: boolean, message: string, type: string) {
    return (dispatch: Dispatch<State>) => {
        dispatch(openNotification({ autoDismiss, message, type }));

        if (autoDismiss) {
            setTimeout(() => {
                dispatch(closeNotification());
            }, ToastTimeout.Duration);
        }
    };
}
