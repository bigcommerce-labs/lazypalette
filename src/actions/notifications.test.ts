import { ToastMessages, ToastType } from './constants';
import * as notifications from './notifications';

describe('notifications actions', () => {
    describe('OPEN_NOTIFICATION', () => {
        it('should macth the correct action', () => {
            const action = notifications.openNotification({
                autoDismiss: true,
                message: ToastMessages.Publish,
                type: ToastType.Success,
            });

            expect(action).toMatchSnapshot();
        });
    });

    describe('CLOSE_NOTIFICATION', () => {
        it('should macth the correct action', () => {
            const action = notifications.closeNotification();

            expect(action).toMatchSnapshot();
        });
    });
});
