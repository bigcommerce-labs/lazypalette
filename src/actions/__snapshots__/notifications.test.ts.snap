// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notifications actions CLOSE_NOTIFICATION should macth the correct action 1`] = `
Object {
  "type": "CLOSE_NOTIFICATION",
}
`;

exports[`notifications actions OPEN_NOTIFICATION should macth the correct action 1`] = `
Object {
  "payload": Object {
    "autoDismiss": true,
    "message": "Changes have been published.",
    "type": "Success",
  },
  "type": "OPEN_NOTIFICATION",
}
`;
